src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -5099,7 +5109,7 @@ private void logon(LogonCommand command) throws SQLServerException {
             } else if (AuthenticationScheme.JAVA_KERBEROS == intAuthScheme) {
                 if (null != impersonatedUserCred) {
                     authentication = new KerbAuthentication(this, currentConnectPlaceHolder.getServerName(),
-                            currentConnectPlaceHolder.getPortNumber(), impersonatedUserCred, isUserCreatedCredential);
+                            currentConnectPlaceHolder.getPortNumber(), impersonatedUserCred, isUserCreatedCredential, useDefaultGSSCredential);
```

Comment:
Sorry for the long delay in review.

Anyways, Is this right? if `impersonatedUserCred` isn't set eg. manually creating/passing in the credential by the user, your new code path will be ignored.

Reply:
Thanks for pointing that out. It should have been:
```
if (null != impersonatedUserCred || useDefaultGSSCredential)
```